signer/signer.go

```diff
@@ -26,6 +26,24 @@ type Subject struct {
 	Hosts []string   `json:"hosts"`
 }
 
+// SignRequest stores a signature request, which contains the hostname,
+// the CSR, optional subject information, and the signature profile.
+type SignRequest struct {
+	Hostname string   `json:"hostname"`
+	Request  string   `json:"certificate_request"`
+	Subject  *Subject `json:"subject,omitempty"`
+	Profile  string   `json:"profile"`
+	Label    string   `json:"label"`
+}
+
+// Root is used to define where the Signer gets its public certificate
+// and private keys for signing.
+type Root struct {
```

Comment:
To sum up the conversation we had, my thinking was along the lines of renaming the fields to Key/CertSpec so as not to introduce confusion later when a root will be adapted to non-file-backed keys, such as via #77.